### PR TITLE
Add ShaderMask benchmark case

### PR DESCRIPTION
**Pull request by JsouLiang**
Add ShaderMask test case for this PR: https://github.com/flutter/engine/pull/32027
cc @flar

### CODE REVIEW

**Comment by wangying3426** [comment]
> I'm sorry I didn't notice that there were 2 PRs from 2 different authors for this same thing, but #100127 already landed so is this PR now a duplicate?

(⊙o⊙)…  @JsouLiang and @wangying3426  work on the same team. 
This test case is added to demonstrates that the saveLayer is need for cached child layer.

File: dev/benchmarks/macrobenchmarks/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -22,6 +22,7 @@ found in the LICENSE file. -->
         android:icon="@mipmap/ic_launcher">
         <activity
             android:name=".MainActivity"
+            android:exported="true"
```
Comment (flar) [inline dev/benchmarks/macrobenchmarks/android/app/src/main/AndroidManifest.xml:25]: I don't see any reason for this flag to be set. These benchmarks aren't exporting any services. Was there a reason this got set here?

File: dev/benchmarks/macrobenchmarks/lib/src/shader_mask_cache.dart [hunk 2]
```diff
@@ -43,6 +43,14 @@ class _ShaderMaskCachePageState extends State<ShaderMaskCachePage>
...
           const SizedBox(height: 10),
           buildShaderMask(1),
+          const SizedBox(height: 10),
+          buildShaderMaskWithBlendMode(BlendMode.modulate, 'BlendMode.modulate'),
+          const SizedBox(height: 10),
+          buildShaderMaskWithBlendMode(BlendMode.clear, 'BlendMode.clear'),
+          const SizedBox(height: 10),
+          buildShaderMaskWithBlendMode(BlendMode.dst, 'BlendMode.dst'),
+          const SizedBox(height: 10),
+          buildShaderMaskWithBlendMode(BlendMode.src, 'BlendMode.src'),
```
Comment (flar) [inline dev/benchmarks/macrobenchmarks/lib/src/shader_mask_cache.dart:53]: Are all of these items visible on our default test device which is a Moto G4?

**Review by flar — CHANGES_REQUESTED** [review]
Mostly questions to verify what I'm seeing.

**Comment by flar** [comment]
> > I'm sorry I didn't notice that there were 2 PRs from 2 different authors for this same thing, but #100127 already landed so is this PR now a duplicate?
> 
> (⊙o⊙)… @JsouLiang and @wangying3426 work on the same team. This test case is added to demonstrates that the saveLayer is need for cached child layer.

This is a benchmark. Nobody ever sees what it looks like as it is run in an anonymous lab. If this is testing that the right output is generated then this should not be a benchmark, it should be a golden test.

File: dev/benchmarks/macrobenchmarks/lib/src/shader_mask_cache.dart (hunk 2 above)
Comment (flar) [reply dev/benchmarks/macrobenchmarks/lib/src/shader_mask_cache.dart:53]: I just fetched this PR and ran this on my G4 and the dst is only partially visible some of the time. The src example never appears.

**Comment by flar** [comment]
This looks like it is modifying an existing benchmark, but that is usually not a practice we encourage because it will disrupt the tracked history of the existing benchmark. We record the performance of all benchmarks over time so we can look for trends.

One reason to modify a benchmark is if we've made breaking changes and it needs to be updated to match the new APIs. That doesn't apply here.

Another reason to modify a benchmark is if it isn't actually exercising the issues that we originally wanted to address (in other words, the execution of the benchmark misses its mark). I don't think that is the case here, but let me know if this applies.

It looks like the existing benchmark was likely measuring something that we want to keep tracking, so it didn't really warrant any updates.

The new code looks like it is testing more conditions beyond the original benchmark, so it will both break the trends we have recorded for the previous version as well as now have 1 benchmark measuring the performance of a couple of cases. If so, then the different conditions warrant being measured in their own benchmarks so that we don't miss regressions when one of the test cases gets faster while the other gets slower - that could hide performance changes that we want to make visible.

I would like to see a better description of what this benchmark was measuring before these changes and what the new changes are measuring and how they differ from or complement each other.

**Pull request closed without merging**